plugins/out_stackdriver/stackdriver.c

```diff
@@ -1937,7 +1937,7 @@ static void set_authorization_header(struct flb_http_client *c,
                                      char *token)
 {
     int len;
-    char header[512];
+    char header[1024];
```

Comment:
According to the Metadata API team, the size could potentially be larger than 1024. See a similar PR at https://github.com/Stackdriver/collectd/pull/192/files

Reply:
Thanks Ling for the reference, they are very useful!

I think 4096 will be a good fit here based on the following considerations:

- There's follow-up PR193 in the collectd repo that revert 65536 back to 4096, with a reference for token size: https://developers.google.com/identity/protocols/oauth2#size, where the current max size for access token is 2048 bytes.

- There's a 4096 bytes buffer when the stackdriver plugin is retrieving token from the metadata server, ref: https://github.com/fluent/fluent-bit/blob/v1.7.3/plugins/out_stackdriver/gce_metadata.c#L72

Reply:
Sorry for the confusion. There was a https://github.com/Stackdriver/collectd/pull/194 that reverted the revert of https://github.com/Stackdriver/collectd/pull/193. Ultimately we decided to leave it as 65536 for the monitoring agent.

The suggestion from the Metadata API team is that:

```
Starting on January 2022, Access token size limit will be set to 12K to match our Public API limits in http://cloud/iam/docs/reference/sts/rest/v1beta/TopLevel/token#response-body.

This does not mean that we are going to start responding with 12KB access tokens. It only means that an access token can be as large as 12KB.

In compliance with http://go/rfc/6749#section-4.2.2 We strongly recommend that customers do not to make any assumptions about the maximum access token size.
```

So something larger than `12KB` as the buffer size might reduce potential issues in the future.

Reply:
thanks for the info @qlingling128

@leonzz In that case, besides the size change, I'd suggest allocating the buffer from the heap instead of using the stack.

Reply:
Thanks Ling and Eduardo. Will make the update to 12KB then:
- when talking to metadata server to fetch token, the http read buffer also need to increase (maybe to 12KB + 256B for other smaller fields);
- It looks to me the extra `snprintf` string copy step can be removed by directly using the sds string: `flb_http_add_header(c, "Authorization", 13, token, flb_sds_len(token));`

Will do the test and update this PR accordingly.